src/main/java/com/teragrep/rlp_03/context/frame/rental/Rental.java

```diff
@@ -0,0 +1,52 @@
+package com.teragrep.rlp_03.context.frame.rental;
+
+import java.util.concurrent.Phaser;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public final class Rental implements AutoCloseable, Consumer<Lease>, Supplier<Lease> {
+
+    private final Phaser phaser;
+    private boolean closed;
```

Comment:
this is extra, should return phaser.isTerminated() ?

Reply:
return phaser.isTerminated: 334d130d91728b1a8209c4d43affee26ed4c2baa